doschema/transform.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of DoSchema.
+# Copyright (C) 2016 CERN.
+#
+# DoSchema is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# DoSchema is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with DoSchema; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Transformation module."""
+
+
+import jsonschema
+import six
+
+
+def transform(schema, uri='', schema_part=None):
```

Comment:
As for the validation we need to be able to provide the `RefResolver`.
We could make two functions:
```python
def resolve_references(schema, uri, ref_resolver=None):
    if ref_resolver is None:
        ref_resolver  = jsonschema.RefResolver(base_uri=uri, referrer=schema)
    _resolve_references_sub(schema, ref_resolver)

def _resolve_references_sub(schema, ref_resolver):
    # use schema as schema_part
```